fastapi_security/basic.py

```diff
@@ -36,3 +53,47 @@ def _make_credentials(
             c if isinstance(c, HTTPBasicCredentials) else HTTPBasicCredentials(**c)
             for c in credentials
         ]
+
+
+class BasicAuthWithDigestValidator:
+    def __init__(self):
+        self._salt = None
+        self._credentials = []
+
+    def init(self, salt: str, credentials: IterableOfHTTPBasicCredentialsDigest):
```

Comment:
```suggestion
    def init(self, credentials: IterableOfHTTPBasicCredentialsDigest, *, salt: str):
```
As per earlier comments.

Reply:
I'm slightly confused: this is the `init` of the `BasicAuthValidator`, but in one of the previous comments you requested to add a class param to be able to select a `BasicAuthWithDigestValidator`, and if we have the latter there is no need to pass `salt` param into `BasicAuthValidator`. Am I missing something here?

Reply:
:+1: